imu_filter_madgwick/src/imu_filter.cpp

```diff
@@ -37,195 +173,125 @@ ImuFilter::~ImuFilter()
 }
 
 void ImuFilter::madgwickAHRSupdate(
-  float gx, float gy, float gz, 
-  float ax, float ay, float az, 
-  float mx, float my, float mz, 
-  float dt)
+    float gx, float gy, float gz,
+    float ax, float ay, float az,
+    float mx, float my, float mz,
+    float dt)
 {
-	float recipNorm;
-	float s0, s1, s2, s3;
-	float qDot1, qDot2, qDot3, qDot4;
-	float hx, hy;
-	float _2q0mx, _2q0my, _2q0mz, _2q1mx, _2bx, _2bz, _4bx, _4bz, _2q0, _2q1, _2q2, _2q3, _2q0q2, _2q2q3, q0q0, q0q1, q0q2, q0q3, q1q1, q1q2, q1q3, q2q2, q2q3, q3q3;
-	float _w_err_x, _w_err_y, _w_err_z;
-
-	// Use IMU algorithm if magnetometer measurement invalid (avoids NaN in magnetometer normalisation)
-  if(!std::isfinite(mx) || !std::isfinite(my) || !std::isfinite(mz))
+  float s0, s1, s2, s3;
+  float qDot1, qDot2, qDot3, qDot4;
+  float _2bz, _2bxy;
+
+  // Use IMU algorithm if magnetometer measurement invalid (avoids NaN in magnetometer normalisation)
+  if (!std::isfinite(mx) || !std::isfinite(my) || !std::isfinite(mz))
   {
-		madgwickAHRSupdateIMU(gx, gy, gz, ax, ay, az, dt);
-		return;
-	}
+    madgwickAHRSupdateIMU(gx, gy, gz, ax, ay, az, dt);
+    return;
+  }
+
+  // Compute feedback only if accelerometer measurement valid (avoids NaN in accelerometer normalisation)
+  if (!((ax == 0.0f) && (ay == 0.0f) && (az == 0.0f)))
+  {
+    // Normalise accelerometer measurement
+    normalizeVector(ax, ay, az);
+
+    // Normalise magnetometer measurement
+    normalizeVector(mx, my, mz);
+
+    // Compensate for magnetic distortion
+    compensateMagneticDistortion(q0, q1, q2, q3, mx, my, mz, _2bxy, _2bz);
+
+    // Gradient decent algorithm corrective step
+    s0 = 0.0;  s1 = 0.0;  s2 = 0.0;  s3 = 0.0;
+    if (useNED_) {
+      // Not optimized version
+      // This code runs as fast as the optimized version when compiled with -O2
```

Comment:
I'd remove those two comment lines; this makes sense in the commit message, but for somebody reading this code for the first time, it's not clear what the "optimized version" is.